cgnet/simplify.py

```diff
@@ -0,0 +1,33 @@
+import networkx as nx
+
+def is_simple_node( graph, node ):
+    """A node is "Simple" if none of the following is true
+    - it has multiple inputs (it joins chains together)
+    - it has no inputs (it's a root node)
+    - it has multiple outputs (it splits chains apart)
+    - it has no outputs (it's a leaf node)
+
+    Keyword arguments:
+    node -- A networkx DiGraph Node
+    """
+    return graph.in_degree(node) == 1 \
+            and graph.out_degree(node)  == 1
+
+def simplify_graph( graph ):
+    """Simplify a CallGraph by collapsing call chains and dropping
```

Comment:
Maybe "coalescing call chains"? Perhaps this language is taken from the go-soaap work, in which case, I'm sorry I didn't think of *le mot juste* sooner! :)